### ISSUE TITLE
Try out Agoric's query unblocking

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#604]
Agoric has work that makes queries no longer block on a mutex while executing a block. This will be huge for osmosis epoch times, and general reliability of service. We should test this out once we upgrade to SDK v0.44.x

SDK work: https://github.com/agoric-labs/cosmos-sdk/pull/114
Tendermint work: https://github.com/tendermint/tendermint/compare/v0.34.13...agoric-labs:Agoric

Thank you @michaelfig for showing this work to me 🙏 

We should make separate branches with these for osmosis, and test them out upon the upgrade to v0.44.

The reason these aren't merged into mainline Tendermint is over concerns with it changing the mutability guarantees for the app.

**Comment by michaelfig** [comment]
Oh, one more thing.  I also had agoric-labs/cosmos-sdk#111 to reject invalid query heights to ensure that the caller knows exactly what height they're getting.  That may be necessary for robustness.

**Comment by michaelfig** [comment]
> The reason these aren't merged into mainline Tendermint is over concerns with it changing the mutability guarantees for the app.

I'm not sure this is true, as the Tendermint change only introduces a new kind of client.  It is still opt-in for the app to use it.  So, I wouldn't expect Cosmos SDK to opt-in necessarily, but I don't see any real obstacle to making it available at the TM layer.

**Comment by michaelfig** [comment]
@ValarDragon please update to  https://github.com/tendermint/tendermint/compare/v0.34.14...agoric-labs:Agoric .  It's based on v0.34.14, and includes a fix for a deadlock during restarting and replaying transactions we found.

Essentially, we need to mark the committingClient as "initialized" as soon as we get a BeginBlock with a nonempty appHash.  Otherwise the replaying DeliverTxs don't ever obtain the RInitLock.

**Comment by ValarDragon** [comment]
I think we should hold off on this
